Exception thrown, PMD plugin disabled and IntelliJ hangs when loading a second project

IntelliJ IDEA 2024.2.4 (Ultimate Edition
PMD Plugin 2.0.2
IntelliJ needs to be force killed.

IntelliJ Plugin 'PMDPlugin' failed to initialize and will be disabled. Please restart the IDE.

java.lang.IllegalArgumentException: Argument for @NotNull parameter 'actionManager' of com/intellij/openapi/actionSystem/DefaultActionGroup.getChildren must not be null
    at com.intellij.openapi.actionSystem.DefaultActionGroup.$$$reportNull$$$0(DefaultActionGroup.java)
    at com.intellij.openapi.actionSystem.DefaultActionGroup.getChildren(DefaultActionGroup.java)
    at com.intellij.plugins.bodhi.pmd.PMDProjectComponent.updateCustomRulesMenu(PMDProjectComponent.java:124)
    at com.intellij.plugins.bodhi.pmd.PMDProjectComponent.initComponent(PMDProjectComponent.java:80)
    at com.intellij.serviceContainer.ComponentInstanceInitializer.createInstance$suspendImpl(ComponentInstanceInitializer.kt:27)
    at com.intellij.serviceContainer.ComponentInstanceInitializer.createInstance(ComponentInstanceInitializer.kt)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1$1.invokeSuspend(LazyInstanceHolder.kt:162)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1$1.invoke(LazyInstanceHolder.kt)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1$1.invoke(LazyInstanceHolder.kt)
    at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:62)
    at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:163)
    at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1.invokeSuspend(LazyInstanceHolder.kt:160)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1.invoke(LazyInstanceHolder.kt)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1.invoke(LazyInstanceHolder.kt)
    at kotlinx.coroutines.intrinsics.UndispatchedKt.startCoroutineUndispatched(Undispatched.kt:27)
    at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:90)
    at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:123)
    at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:52)
    at kotlinx.coroutines.BuildersKt.launch(Unknown Source)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.initialize(LazyInstanceHolder.kt:145)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.access$initialize(LazyInstanceHolder.kt:13)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.tryInitialize(LazyInstanceHolder.kt:135)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.getInstance(LazyInstanceHolder.kt:95)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.getInstanceInCallerContext$suspendImpl(LazyInstanceHolder.kt:87)
    at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.getInstanceInCallerContext(LazyInstanceHolder.kt)
    at com.intellij.platform.instanceContainer.internal.UtilKt.preloadAllInstances(util.kt:47)
    at com.intellij.serviceContainer.ComponentManagerImpl.createComponentsNonBlocking(ComponentManagerImpl.kt:512)
    at com.intellij.openapi.project.impl.ProjectManagerImplKt$initProject$2$3.invokeSuspend(ProjectManagerImpl.kt:1256)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
    at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:608)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:873)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:763)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:750)

Similar when starting-up:

Cannot create com.intellij.plugins.bodhi.pmd.PMDProjectComponent

com.intellij.ide.plugins.StartupAbortedException: Fatal error initializing plugin PMDPlugin
	at com.intellij.serviceContainer.ComponentManagerImplKt.handleComponentError(ComponentManagerImpl.kt:1453)
	at com.intellij.serviceContainer.ComponentInstanceInitializer.createInstance$suspendImpl(ComponentInstanceInitializer.kt:41)
	at com.intellij.serviceContainer.ComponentInstanceInitializer.createInstance(ComponentInstanceInitializer.kt)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1$1.invokeSuspend(LazyInstanceHolder.kt:162)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1$1.invoke(LazyInstanceHolder.kt)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1$1.invoke(LazyInstanceHolder.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:62)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:163)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1.invokeSuspend(LazyInstanceHolder.kt:160)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1.invoke(LazyInstanceHolder.kt)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder$initialize$1.invoke(LazyInstanceHolder.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startCoroutineUndispatched(Undispatched.kt:27)
	at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:90)
	at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:123)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:52)
	at kotlinx.coroutines.BuildersKt.launch(Unknown Source)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.initialize(LazyInstanceHolder.kt:145)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.access$initialize(LazyInstanceHolder.kt:13)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.tryInitialize(LazyInstanceHolder.kt:135)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.getInstance(LazyInstanceHolder.kt:95)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.getInstanceInCallerContext$suspendImpl(LazyInstanceHolder.kt:87)
	at com.intellij.platform.instanceContainer.internal.LazyInstanceHolder.getInstanceInCallerContext(LazyInstanceHolder.kt)
	at com.intellij.platform.instanceContainer.internal.UtilKt.preloadAllInstances(util.kt:47)
	at com.intellij.serviceContainer.ComponentManagerImpl.createComponentsNonBlocking(ComponentManagerImpl.kt:512)
	at com.intellij.openapi.project.impl.ProjectManagerImplKt$initProject$2$3.invokeSuspend(ProjectManagerImpl.kt:1256)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:608)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:873)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:763)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:750)
Caused by: com.intellij.diagnostic.PluginException: Argument for @NotNull parameter 'actionManager' of com/intellij/openapi/actionSystem/DefaultActionGroup.getChildren must not be null [Plugin: PMDPlugin]
	... 32 more
Caused by: java.lang.IllegalArgumentException: Argument for @NotNull parameter 'actionManager' of com/intellij/openapi/actionSystem/DefaultActionGroup.getChildren must not be null
	at com.intellij.openapi.actionSystem.DefaultActionGroup.$$$reportNull$$$0(DefaultActionGroup.java)
	at com.intellij.openapi.actionSystem.DefaultActionGroup.getChildren(DefaultActionGroup.java)
	at com.intellij.plugins.bodhi.pmd.PMDProjectComponent.updateCustomRulesMenu(PMDProjectComponent.java:124)
	at com.intellij.plugins.bodhi.pmd.PMDProjectComponent.initComponent(PMDProjectComponent.java:80)
	at com.intellij.serviceContainer.ComponentInstanceInitializer.createInstance$suspendImpl(ComponentInstanceInitializer.kt:27)
	... 30 more

--- a/src/main/java/com/intellij/plugins/bodhi/pmd/PMDProjectComponent.java
+++ b/src/main/java/com/intellij/plugins/bodhi/pmd/PMDProjectComponent.java
@@ -118,10 +118,11 @@ public class PMDProjectComponent implements ProjectComponent, PersistentStateCom
      * Now for > 1 projects open, merge the rule sets of shared actions (menu) and current project
      */
     void updateCustomRulesMenu() {
-            PMDCustom actionGroup = (PMDCustom) ActionManager.getInstance().getAction("PMDCustom");
+        ActionManager actionManager = ActionManager.getInstance();
+        PMDCustom actionGroup = (PMDCustom) actionManager.getAction("PMDCustom");
             if (numProjectsOpen.get() != 1) {
                 // merge actions from menu and from settings to not lose any when switching between projects
-                AnAction[] currentActions = actionGroup.getChildren((ActionManager)null);
+                AnAction[] currentActions = actionGroup.getChildren(null, actionManager);
                 Set<String> ruleSetPathsFromMenu = new LinkedHashSet<>();
                 for (AnAction action : currentActions) {
                     if (action.getSynonyms().size() == 1) {

--- a/src/main/java/com/intellij/plugins/bodhi/pmd/actions/PreDefinedMenuGroup.java
+++ b/src/main/java/com/intellij/plugins/bodhi/pmd/actions/PreDefinedMenuGroup.java
@@ -19,13 +19,10 @@ import java.util.Properties;
  * @author bodhi
  * @version 1.0
  */
-public class PreDefinedMenuGroup extends ActionGroup {
+public class PreDefinedMenuGroup extends DefaultActionGroup {
 
     // A string that represents all the rulesets as comma separated value.
     private static String allRules = "";
-
-    //All the children of this group
-    private DefaultActionGroup children = new DefaultActionGroup("Pre Defined", true);
 
     private PMDProjectComponent component;
 
@@ -50,9 +47,7 @@ public class PreDefinedMenuGroup extends ActionGroup {
             String[] rulesetFilenames = props.getProperty(RULESETS_FILENAMES_KEY).split(PMDInvoker.RULE_DELIMITER);
 
             //We have 'All' rules in addition to the rulesets
-            //First one is 'All'
-            children.removeAll();
-            children.add(action);
+            add(action);
 
             for (int i=0; i < rulesetFilenames.length; ++i) {
                 final String ruleFileName = rulesetFilenames[i];
@@ -65,7 +60,7 @@ public class PreDefinedMenuGroup extends ActionGroup {
                         getComponent().setLastRunActionAndRules(e, ruleFileName, false);
                     }
                 };
-                children.add(ruleAction);
+                add(ruleAction);
             }
         } catch (Exception e) {
             throw new RuntimeException(e);
@@ -78,10 +73,6 @@ public class PreDefinedMenuGroup extends ActionGroup {
             return Thread.currentThread().getContextClassLoader().getResourceAsStream(RULESETS_PROPERTY_FILE);
         }
         return resourceAsStream;
-    }
-
-    public AnAction[] getChildren(@Nullable AnActionEvent event) {
-        return new AnAction[]{this.children};
     }
 
     public void setComponent(PMDProjectComponent component) {

--- a/src/main/resources/META-INF/plugin.xml
+++ b/src/main/resources/META-INF/plugin.xml
@@ -22,9 +22,9 @@
       <add-to-group group-id="ChangesViewPopupMenu" anchor="last"/>
       <add-to-group group-id="EditorPopupMenu" anchor="last"/>
 
-      <action class="com.intellij.plugins.bodhi.pmd.actions.PreDefinedMenuGroup" id="PMDPredefined"
+      <group id="PMDPredefined" class="com.intellij.plugins.bodhi.pmd.actions.PreDefinedMenuGroup"
               popup="true" text="Pre Defined"/>
-      <action id="PMDCustom" class="com.intellij.plugins.bodhi.pmd.actions.PMDCustom"
+      <group id="PMDCustom" class="com.intellij.plugins.bodhi.pmd.actions.PMDCustom"
               popup="true" text="Custom Rules"/>
     </group>
 

